### ISSUE TITLE
Const Distance between earth and sun 

### ISSUE THREAD

**Issue by jindal12ketan** [mourner/suncalc#175]
`SunCalc.getMoonIllumination = function (date) {

    var d = toDays(date || new Date()),
        s = sunCoords(d),
        m = moonCoords(d),

        sdist = 149598000, // distance from Earth to Sun in km

        phi = acos(sin(s.dec) * sin(m.dec) + cos(s.dec) * cos(m.dec) * cos(s.ra - m.ra)),
        inc = atan(sdist * sin(phi), m.dist - sdist * cos(phi)),
        angle = atan(cos(s.dec) * sin(s.ra - m.ra), sin(s.dec) * cos(m.dec) -
                cos(s.dec) * sin(m.dec) * cos(s.ra - m.ra));

    return {
        fraction: (1 + cos(inc)) / 2,
        phase: 0.5 + 0.5 * inc * (angle < 0 ? -1 : 1) / Math.PI,
        angle: angle
    };
};`

Why is the sdist constant, doesn't it change with date and time?

**Comment by lulunac27a** [comment]
This number is the distance of 1 AU (astronomical unit) in kilometers.

**Comment by jindal12ketan** [comment]
That's correct, but my question is that the distance between the sun and Earth always changes with time, so we need to keep that distance dynamic for accuracy.

**Comment by lulunac27a** [comment]
Earth eccentricity is 0.0167, perihelion is January 4

**Comment by zedxxx** [comment]
I can assume that using a distance of 1 AU provides a sufficiently accurate result. 

For experimental purposes, plug the Aphelion distance of 152,098,233 km (around July 5) and the Perihelion distance of 147,098,291 km (around January 3) into the formula, and compare the error margin relative to 1 AU. I don’t believe the deviation will be significant. 

I’ll also note that a more precise value for 1 AU = 149,597,870.7 km could be used.

**Comment by mourner** [comment]
Good question — sorry for such a late reply. `sdist` is the mean Earth–Sun distance (1 AU), and yes, the real distance varies by ±1.67% over the year (perihelion ~Jan 3, aphelion ~Jul 5). But that variation barely touches the result. The phase angle uses `atan2(sdist·sin φ, m.dist − sdist·cos φ)`, and because the Moon is so much closer to Earth than the Sun, sweeping `sdist` across the full perihelion→aphelion range moves the illuminated `fraction` by at most ~2×10⁻⁵ (0.002%) — worst case near quarter phase:

```
phi=1.5  fraction=0.465912  Δ(perihelion)=+2.2e-5  Δ(aphelion)=-2.1e-5
```

That's far below SunCalc's accuracy floor, so keeping it constant is a deliberate (and harmless) simplification rather than an oversight. @lulunac27a and @zedxxx are both correct above. Closing — but thanks for the careful read.
